src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -443,8 +444,13 @@ private static void CancelToken(ChangeTokenInfo matchInfo)
                 TaskScheduler.Default);
         }
 
-        internal static void RaiseChangeEvents(object state)
+        internal static void RaiseChangeEvents(object? state)
```

Comment:
should state be non-null? I'm seeing couple of places in this PR which don't expect null

Reply:
This (`Action<object?>`) is required by event contract. I guess, it would be better to define it as non-nullable and cast it to `Action<object?>` when subscribing to event. Didn't thought of that before.

Reply:
Will this be better?

![image](https://user-images.githubusercontent.com/6609929/129610207-5ed493bf-b162-46f3-ac75-783d9791fe9d.png)

Reply:
nullable state seems fine in that case